R/stats-mlm-tidiers.R

```diff
@@ -0,0 +1,99 @@
+#' @templateVar class mlm
+#' @template title_desc_tidy
+#'
+#' @param x An `mlm` object created by [stats::lm()] with a matrix as the
+#'   response.
+#' @template param_confint
+#' @template param_quick
+#' @template param_unused_dots
+#'
+#' @evalRd return_tidy(regression = TRUE)
+#'
+#' @details In contrast to `lm` object (simple linear model), tidy output for
+#'   `mlm` (multiple linear model) objects contain an additional column
+#'   `response`.
+#'
+#'   If you have missing values in your model data, you may need to refit
+#'   the model with `na.action = na.exclude`.
+#'
+#' @examples
+#'
+#' mod <- lm(cbind(mpg, disp) ~ wt, mtcars)
+#' tidy(mod, conf.int = TRUE)
+#' @export
+#' @seealso [tidy()]
+#' @family lm tidiers
+#' @export
+
+tidy.mlm <- function(x,
+                     conf.int = FALSE,
+                     conf.level = .95,
+                     ...) {
+
+  # adding other details from summary object
+  s <- summary(x)
+  ret <- tidy.summary.mlm(s)
+
+  # adding confidence intervals
+  if (isTRUE(conf.int)) {
+
+    # S3 method for computing confidence intervals for `mlm` objects was
+    # introduced in R 3.5
+    CI <- tryCatch(
```

Comment:
This isn't quite what I meant. This tries to get a CI and doesn't add the CI if things break. This makes the tidier sort of magical, in that the user doesn't know what columns they are going to get back. If the user requests tidiers and can't get them, we should throw an informative error, rather than returning the standard columns other than `conf.low` and `conf.high`.

Reply:
No, this always returns CI, either using `stats::confint` or using the custom `confint` method in `broom` (which has reintroduced to support `3.4.0`).